app/main.py

```diff
@@ -1,3 +1,55 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float) -> Distance:
+        if isinstance(other, Distance):
+            value = other.km
+        else:
+            value = other
+        return Distance(self.km + value)
+
+    def __iadd__(self, other: Distance | float) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: int) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: int) -> Distance:
+        return Distance(round(self.km / other, 2))
+
+    def __eq__(self, other: int) -> bool:
+        return self.km == other
+
+    def __gt__(self, other: Distance | float) -> bool:
+        return (self.km > other.km) \
+            if isinstance(other, Distance) \
+            else self.km > other
```

Comment:
This implementation of the function will work, but using a one-liner if, then splitting it doesn't look very good. It's better to do it like this:
```suggestion
def __gt__(self, other: Distance | float) -> bool:
    if isinstance(other, Distance):
        return self.km > other.km
    return self.km > other

```